genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java

```diff
@@ -155,12 +157,72 @@ public Cluster selectCluster(
         final long selectStart = System.nanoTime();
         log.debug("Called");
         final Set<Tag> tags = Sets.newHashSet();
+
         try {
-            if (this.isConfigured.get() && this.script.get() != null) {
-                log.debug("Evaluating script for job {}", jobRequest.getId().orElse("without id"));
-                final Bindings bindings = new SimpleBindings();
-                // TODO: For now for backwards compatibility with balancer scripts continue writing Clusters out in
-                //       V3 format. Change to V4 once stabalize a bit more
+            final CompiledScript script = this.loader.get();
+            if (script == null) {
+                log.debug("Script not configured");
+                tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_NOT_CONFIGURED));
+                return null; // Could throw if a script is expected when invoked
+            }
+
+            final String clusterId = this.evaluator.evaluate(script, jobRequest, clusters);
+
+            if (clusterId == null) {
+                tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_NO_PREFERENCE));
+                log.debug("Script returned null, no preference");
+                return null;
+            }
+
+            for (final Cluster cluster : clusters) {
+                if (clusterId.equals(cluster.getId())) {
+                    tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_FOUND));
```

Comment:
Maybe we should tag with the cluster id selected as well?

Reply:
Would be nice to have.
Tho then we should also do the same in the class that uses this component (`JobSpecService`), since more than one `LoadBalancer` may be used.

Reply:
Added a commit that adds metrics tags